skcosmo/feature_selection/_greedy.py

```diff
@@ -0,0 +1,238 @@
+            if not 0 < self.n_features_to_select <= 1:
+                raise ValueError(error_msg)
+            n_iterations = int(n_features * self.n_features_to_select)
+        else:
+            raise ValueError(error_msg)
+
+        if warm_start:
+            if self.n_selected_ == 0:
+                raise ValueError(
+                    "Cannot fit with warm_start=True without having been previously initialized"
+                )
+            self._continue_greedy_search(X, y, n_iterations)
+        else:
+            self.n_selected_ = 0
+            self._init_greedy_search(X, y, n_iterations)
+
+        n_iterations -= self.n_selected_
+
+        for n in self.report_progress(range(n_iterations)):
+
+            new_feature_idx = self._get_best_new_feature(self.scoring, X, y)
+            if new_feature_idx is not None:
+                self._update_post_selection(X, y, new_feature_idx)
+            else:
+                warnings.warn(
+                    f"Score threshold of {self.score_threshold} reached."
+                    f"Terminating search at {self.n_selected_} / {self.n_features_to_select}."
+                )
+                self.X_selected_ = self.X_selected_[:, :n]
+                self.selected_idx_ = self.selected_idx_[:n]
+                self._postprocess(X, y)
+                return self
+
+        self._postprocess(X, y)
+        return self
+
+    def _init_greedy_search(self, X, y, n_to_select):
+        """ Initializes the search. Prepares an array to store the selected features. """
+
+        self.X_selected_ = np.zeros((X.shape[0], n_to_select), float)
+        self.selected_idx_ = np.zeros((n_to_select), int)
+
+    def _continue_greedy_search(self, X, y, n_to_select):
+        """ Continues the search. Prepares an array to store the selected features. """
+
+        self.X_selected_ = np.pad(
+            self.X_selected_,
+            [(0, 0), (0, n_to_select - self.n_selected_)],
+            "constant",
+            constant_values=0.0,
+        )
+        old_idx = self.selected_idx_.copy()
+        self.selected_idx_ = np.zeros((n_to_select), int)
+        self.selected_idx_[: self.n_selected_] = old_idx
+
+    def _get_best_new_feature(self, scorer, X, y):
+
+        scores = scorer(X, y)
+
+        if self.score_threshold is not None and max(scores) < self.score_threshold:
+            return None
+        else:
+            return np.argmax(scores)
+
+    def _update_post_selection(self, X, y, last_selected):
+        """
+        Saves the most recently selected feature and increments the feature counter
+        """
+
+        self.X_selected_[:, self.n_selected_] = X[:, last_selected]
+        self.selected_idx_[self.n_selected_] = last_selected
+        self.n_selected_ += 1
+
+    def _get_support_mask(self):
+        check_is_fitted(self, ["support_"])
+        return self.support_
+
+    def _postprocess(self, X, y):
+        """ Post-process X and / or y when selection is finished """
```

Comment:
The docstring claims that X and/or y are post-processed, but the code below only appears to rely on X. How does y fit in?

Reply:
This is a method meant to be overridden when necessary, and should take y for PCov formulations.